fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/common/AddColumn.java

```diff
@@ -38,7 +43,19 @@ public AddColumn(String schemaName, String tableName, ColumnBase column) {
     @Override
     public void run(IDatabaseTranslator translator, Connection c) {
         String qname = DataDefinitionUtil.getQualifiedName(schemaName, tableName);
-        String ddl = "ALTER TABLE " + qname + " ADD COLUMN " + columnDef(column);
+
+        // DatabaseTypeAdapter is needed to find the correct data type for the column.
+        IDatabaseTypeAdapter dbAdapter = null;
+        String driveClassName = translator.getDriverClassName();
+        if (driveClassName.contains("db2")) {
+            dbAdapter = new Db2Adapter((IConnectionProvider)null);
```

Comment:
why is a null being cast?

Reply:
this code doesn't really fit the pattern.

Reply:
added new constructor for the db adapters to make the codes better.